app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1225,14 +1345,220 @@ public void onDateSet(DatePicker view, int year, int month, int day) {
+
+
+        public TimePickerFragment(Context context, View showRemindersLayout, Button addReminderBtn) {
+
+            this.context = context;
+            this.showRemindersLayout = showRemindersLayout;
+            this.addReminderBtn = addReminderBtn;
+
+            daysTextView = showRemindersLayout.findViewById(R.id.daysTextView);
+            dateReminderView =showRemindersLayout.findViewById(R.id.dateReminderView);
+            timeReminderTextView =showRemindersLayout.findViewById(R.id.timeReminderTextView);
+
+        }
+
+        @NonNull
+        @Override
+        public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+
+            calendar = Calendar.getInstance();
+
+            Date date = (Date) timeReminderTextView.getTag();
+
+            if (date != null) {
+                calendar.setTime(date);
+            }
+
+
+            hour = calendar.get(Calendar.HOUR_OF_DAY);
+            minute = calendar.get(Calendar.MINUTE);
+
+            return new TimePickerDialog(getActivity(), this, hour, minute, false);
+        }
+
+        @Override
+        public void onTimeSet(TimePicker timePicker, int mhour, int mminute) {
+
+            calendar.set(Calendar.HOUR_OF_DAY, mhour);
+            calendar.set(Calendar.MINUTE, mminute);
+
+            hour = mhour;
+            minute = mminute;
+
+            Date date = new Date(calendar.getTimeInMillis());
+            timeReminderTextView.setTag(date);
+
+            delay.set(Calendar.HOUR_OF_DAY,mhour);
+            delay.set(Calendar.MINUTE,mminute);
+
+            Log.i(TAG, "onTimeSet: timeInMillis "+delay.getTimeInMillis());
+
+            showReminders(addReminderBtn);
+            hasReminderChanged = true;
+
+        }
+
+        private void showReminders(Button addReminderBtn) {
+
+
+
+            Log.i(TAG, "showReminders: days ="+days);
+
+            daysTextView.setText(String.valueOf(days));
+            dateReminderView.setText(formattedDate);
+            timeReminderTextView.setText(formatTime(hour,minute));
+
+            showRemindersLayout.setVisibility(View.VISIBLE);
+            addReminderBtn.setText(R.string.edit_reminder);
+
+        }
+
+    }
+
+
+        private static String formatTime(int mhour, int mminute) {
+            String sHour,sMinute;
+            String sTime , am_pm;
+
+            if (mhour >= 12){
+
+                am_pm = "PM";
+
+                if (mhour>12) {
+                    mhour -=12;
+                }
+            }
+            else {
+                am_pm = "AM";
+
+                if (mhour == 0) {
+
+                    mhour = 12;
+                }
+            }
+            if (mminute <10) {
+
+                sMinute = "0"+mminute;
+
+            }
+            else {
+                sMinute = String.valueOf(mminute);
+            }
+            sHour = mhour+":";
+            sMinute +=" "+am_pm;
+            sTime = sHour+sMinute;
+
+            return sTime ;
+        }
```

Comment:
I'm sure Android has a built-in library to format time according to user locale. https://developer.android.com/reference/java/time/format/DateTimeFormatter#ofLocalizedTime(java.time.format.FormatStyle) seems like what you are looking for at first glace.